src/pages/postgraphile/make-wrap-resolvers-plugin-audit-and-soft-delete.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: page
+path: /postgraphile/make-wrap-resolvers-plugin/
+title: makeWrapResolversPlugin audit & soft-delete example (graphile-utils)
+---
+### Example plugin for audit columns & soft delete (created_at, updated_at, deleted_at) 
+
+The below code contains a single plugin that detects if the mutation target table contains
+`created_at` or `updated_at` columns and overrides the query payload values with current date.
+
+The same plugin also detects if the mutation target table contains a `deleted_at` column and 
+overrides the delete command with custom sql query (which, in this case, is `update [table] set deleted_at = now()`).
+
+> NOTE to Benjie & the team: This example is provided with my limited knowledge as I have just started 
+> plugin development and with this disclaimer I can note that by using lodash and my solution for returning
+> the Delete mutation result to client is a hack that shows my limited knowledge ✌️😉
+>
+> ps. Please modify this page however you want to fit more generic usage of Postgraphile.    
+
+```js
+const _ = require('lodash')
+const { makeWrapResolversPlugin } = require('graphile-utils')
+
+module.exports = makeWrapResolversPlugin(
+  context => {
+    if (context.scope.isRootMutation) {
+      return { scope: context.scope }
+    }
+    return null;
+  },
+  ({ scope }) => async (resolver, user, args, context, _resolveInfo) => {
+
+    const { pgClient } = context
+
+    if (
+      scope.isPgUpdateMutationField &&
+      // check if pgFieldIntrospection contains "updated_at" column
+      _.find(_.get(scope, 'pgFieldIntrospection.attributes', []), {
+        kind: 'attribute',
+        name: 'updated_at'
+      })
+    ) {
+      // modify input.patch with updatedAt value (eg. timestamp)
+      _.set(args, 'input.patch.updatedAt', (new Date()).toISOString())
+    } else if (
+      scope.isPgCreateMutationField &&
+      // check if pgFieldIntrospection contains "created_at" column
+      _.find(_.get(scope, 'pgFieldIntrospection.attributes', []), {
+        kind: 'attribute',
+        name: 'created_at'
+      })
+    ) {
+      let objectName = _.first(_.keys(args.input))
+      _.set(args, `input.${objectName}.createdAt`, (new Date()).toISOString())
+    } else if (
+      scope.isPgDeleteMutationField &&
+      _.find(_.get(scope, 'pgFieldIntrospection.attributes', []), {
+        kind: 'attribute',
+        name: 'deleted_at'
+      })
+    ) {
+
+      // hack: reconstructing schema + table name from pgFieldIntrospection. 
+      const table = _.get(scope, 'pgFieldIntrospection.namespaceName') + '.' + _.get(scope, 'pgFieldIntrospection.name')
+      // hack: reconstructing return payload GraphqlType from table name.
+      const gqlName = '@' + _.camelCase(_.get(scope, 'pgFieldIntrospection.name'))
```

Comment:
This is definitely the wrong way to do this. If you use a GraphQL alias then the mutation will break - you should basically never manually generate an `@`-prefix.